authorize_helper.go

```diff
@@ -131,10 +131,6 @@ func isLoopbackURI(requested *url.URL, registeredURI string) bool {
 		return false
 	}
 
-	if registered.Scheme != "http" || !isLoopbackAddress(registered.Host) {
```

Comment:
This is redundant. So I removed it.

Reply:
While that's currently true, removing explicit checks for implicit ones opens up a big opportunity for regression issues. Given that this is security sensitive, and that you want to have this removed, please either write a test covering this exact case or point to the test case that already covers this logic. Thanks! :)

Reply:
But this is logically redundant. It is literally the opposite than the next case. It is De Morgan law.

Compare:

```
registered.Scheme != "http" || !isLoopbackAddress(registered.Host)
```

and:

```
requested.Scheme == "http" && isLoopbackAddress(requested.Host)
```

Reply:
Oh, you just want a test case where using `http` scheme returns false and using a `http` scheme without loopback address returns false? I [think those are already covered](https://github.com/ory/fosite/blob/master/authorize_helper_test.go#L229-L246).

Reply:
Exactly!

That test tests `IsRedirectURISecure` which does not use `isLoopbackURI` so I think this is neither covered explicitly nor implicitly.